src/sledo/design_evaluator.py

```
@@ -0,0 +1,351 @@
+        model : MooseModel
+            A catbird MooseModel capable of updating parameters and writing a
+            MOOSE input file.
+        working_dir : Path | str, optional
+            Path to the working directory to use for storing modified MOOSE
+            input files (.i) and running MOOSE, by default Path.cwd().
+        config_path : Path | str, optional
+            Path to the config file containing the required paths to run MOOSE,
+            by default 'moose_config.json' in the sledo root folder.
+        run_options : dict, optional
+            Dict of options for running the simulation, by default
+            { "n_tasks": 1, "n_threads": 4, "redirect_out": False }.
+        """
+        self._metrics = metrics
+        self._model = model
+        self.working_dir = Path(working_dir)
+        self.config_path = Path(config_path)
+        self.run_options = run_options
+
+    @property
+    def metrics(self):
+        return self._metrics
+
+    def add_metric_postprocessor(
+        self,
+        metric: str,
+        variable_name: str,
+        postprocessor_type: str = "ElementExtremeValue",
+        **postprocessor_kwargs,
+    ):
+        """Add a Postprocessor object to the MOOSE input file with a name
+        matching an optimisation metric. Available postprocessors here:
+        https://mooseframework.inl.gov/syntax/Postprocessors/index.html
+
+        Parameters
+        ----------
+        metric : str
+            Name of metric to add a postprocessor for, must match exactly the
+            name of an optimisation metric so that it may be reported to the
+            optimiser.
+        variable_name : str
+            Name of variable upon which the postprocessor will act.
+        postprocessor_type : str, optional
+            Type of postprocessor to use, by default "ElementExtremeValue".
+        """
+        if metric not in self._metrics:
+            print(
```

I would raise an error here, don't want a poor user to run a whole massive sweep and find out there post-processor didn't setup correctly